lib/spack/docs/build_settings.rst

```diff
@@ -209,6 +209,170 @@ Specific limitations include:
   then Spack will not add a new external entry (``spack config blame packages``
   can help locate all external entries).
 
+
+.. _manually-adding-external-packages:
+
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Manually Adding External Packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This section discusses how to add external packages to the ``packages.yaml``
+file when modules depend on other modules or when Spack is unable to compute
+the prefix. When using modules, Spack will attempt to parse the ``module show``
+output in order to determine all relevant settings and variables. Reasons why
+this may not work automatically are
+
+* the wrong package name,
+* missing module dependencies,
+* nonstandard paths, and
+* the use of metamodules, i.e., modules whose only purpose is to load other
+  modules.
+
+For certain libraries there exist multiple implementations (e.g., MPI). If you
```

Comment:
This paragraph feels out of place: it has no prior introduction. Should it actually be a 5th bullet point (in addition to the 4 above)? A separate section?